File: 144. Binary Tree Preorder Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void trav(TreeNode root, ArrayList<Integer> arr){
        if(root == null) return;

        arr.add(root.val);
        trav(root.left, arr);
        trav(root.right, arr);
    }
    public List<Integer> preorderTraversal(TreeNode root) {
        ArrayList<Integer> arr = new ArrayList<>();
        trav(root, arr);

        return arr;
    }
}

File: 144. Binary Tree Preorder Traversal.cpp
class Solution {
public:
    void trav(TreeNode* root, vector<int>& arr) {
        if (root == nullptr) return;

        arr.push_back(root->val);
        trav(root->left, arr);
        trav(root->right, arr);
    }

    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> arr;
        trav(root, arr);
        return arr;
    }
};
